src/runtime/crt/host/main.cc

```diff
@@ -93,6 +94,20 @@ tvm_crt_error_t TVMPlatformTimerStop(double* elapsed_time_seconds) {
   g_utvm_timer_running = 0;
   return kTvmErrorNoError;
 }
+
+static_assert(RAND_MAX >= (1 << 8), "RAND_MAX is smaller than acceptable");
+unsigned int random_seed = 0;
+tvm_crt_error_t TVMPlatformGenerateRandom(uint8_t* buffer, size_t num_bytes) {
+  if (random_seed == 0) {
+    random_seed = (unsigned int)time(NULL);
+  }
+  for (size_t i = 0; i < num_bytes; ++i) {
+    int random = rand_r(&random_seed);
+    buffer[i] = (uint8_t)random;
```

Comment:
I see that zephyr one has a better version -- as in all the 4 bytes of random is being used with a prologue to handle the tail. I think that is a better implementation. Since you have it already, could anything be done to make that part common in the code base and use it here as well ?

Reply:
this is one of those really tricky areas in embedded. you could imagine factoring out the logic for that into a library function, but you would be implicitly relying on:
1. link-time optimization to ensure the common impl gets deleted when not used
2. a function pointer, which may defeat any inlining
3. an assumption that most platforms can generate 32-bit random numbers, from the POV of making the conceptual complexity worth adding

so for "glue-like" logic like this, unfortunately my opinion is that it's best to copy-paste the stub that fits your particular situation. an embedded transpiler would really help cases like this, but that drags in another debate :)

I think given the main difference is a linear speedup in random blob generation on host simulation, and rand_r should not syscall, i'm going to defer any potential change here to the next PR (which enables autotuning and relies on this RNG a lot more).

Reply:
Well, I was actually refferring to make it go to common util header and include it where it will be used, thus common impl is explicitly included where its needed -- ( if none uses it then it should not get included ? ).

Anyway, copy-paste stub might be ok for now, but curious why this one has to be different from to zephyr one ?

Reply:
ah i see, yeah that would work. hmm, I think if we had the ability to template the random block type (I.e. uint32_t right now) without a C macro i'd do it, but maybe it's okay to leave it as is for now.

the main difference between the zephyr one and the host one is that the underlying rng function is different. the host uses `rand_r` and zephyr uses `sys_rand32_get`. note that this main.cc is not compiled using zephyr but rather just directly against the system headers, and it uses posix pipes to communicate.

Reply:
Alright, I think its OK to use the different rng function given the compilation flows. However, I was referring to difference in handling (and absence of it) the tail. Would it be possible to have that kind of code with the different rng here as well ?

Reply:
well in this case we don't have a tail to worry about because we generate 32-bit random integers but throw away the upper 24 bits, and write the lower 8 bits to successive bytes in `buffer`. we are throwing away randomness but it's pseudorandomness anyhow. I may revisit this if generating random tensors is slow in the AutoTVM PR.

Reply:
The thing Im struggling to understand is why throwing away 24 bit is relatively not important compared to the other. Is it because these runtimes are intended to run on different compute ability ? -- I thought vanilla crt is also intended run on constrained devices.

Reply:
ah I see. I am being a little bit lazy here as right now, this mainly serves (in the tvm repo) to support tests/python/unittest/test_crt.py. I can implement a better RNG here if it would help.

Reply:
My concern is to keep the crt flow upto the standard as zephyr example you are showing here :). 
If its too much trouble, I would think the zephyr implementation is superior (might be a delta difference though).